src/pages/Register.tsx

```diff
@@ -0,0 +1,160 @@
+import { createUserWithEmailAndPassword } from 'firebase/auth';
+import React, { useState } from 'react';
+import { auth, db } from '../firebase/index'
+import { setDoc, doc } from 'firebase/firestore';
+import { useNavigate } from 'react-router-dom';
+
+export function Register() {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [fname, setFname] = useState("");
+    const [lname, setLname] = useState("");
+    const [phone, setPhone] = useState("");
+    const [confirmPassword, setconfirmPassword] = useState("");
+
+    const navigate = useNavigate();
+
+    const handleRegister = async (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+
+        // Validación del nombre
+        const nameRegex = /^[A-Za-z]+( [A-Za-z]+)?$/;
+        if (!nameRegex.test(fname.trim())) {
+            console.error("Invalid name format. The name must contain only these letters, and have at most one space.");
+            return;
+        }
+
+        // Validación de formato de correo
+        const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+        if (!emailRegex.test(email)) {
+            console.error("Invalid email format or the email contains spaces");
+            return;
+        }
+
+        // Validación de la contraseña
+        const passwordRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[A-Za-z\d]{6,}$/;
+        if (!passwordRegex.test(password)) {
+            console.error("Password must be at least 6 characters long and include at least one uppercase letter, one lowercase letter, and one number.");
+            return;
+        }
+
+        // Validación de contraseñas que coincidan
+        if (password !== confirmPassword) {
+            console.error("Passwords do not match");
+            return;
+        }
+
+        try {
+            await createUserWithEmailAndPassword(auth, email, password);
+            const user = auth.currentUser;
+            if (user) {
+                await setDoc(doc(db, "Users", user.uid), {
```

Comment:
utilizar minúsculas y en plural para apuntar a la collection de firebase, ejemplo:
```suggestion
                await setDoc(doc(db, "users", user.uid), {
```